src/sentry/replays/endpoints/organization_replay_selector_index.py

```diff
@@ -0,0 +1,214 @@
+from __future__ import annotations
+
+from datetime import datetime
+from typing import Any, List, Optional
+
+from rest_framework.exceptions import ParseError
+from rest_framework.request import Request
+from rest_framework.response import Response
+from snuba_sdk import (
+    Column,
+    Condition,
+    Direction,
+    Entity,
+    Function,
+    Granularity,
+    Limit,
+    Offset,
+    Op,
+    OrderBy,
+    Query,
+)
+from snuba_sdk import Request as SnubaRequest
+
+from sentry import features
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.organization import NoProjects, OrganizationEndpoint
+from sentry.api.event_search import SearchConfig
+from sentry.api.paginator import GenericOffsetPaginator
+from sentry.models.organization import Organization
+from sentry.replays.lib.query import Number, QueryConfig, get_valid_sort_commands
+from sentry.replays.query import Paginators, make_pagination_values
+from sentry.replays.validators import ReplayValidator
+from sentry.utils.snuba import raw_snql_query
+
+
+@region_silo_endpoint
+class OrganizationReplaySelectorIndexEndpoint(OrganizationEndpoint):
+    def get_replay_filter_params(self, request, organization):
+        filter_params = self.get_filter_params(request, organization)
+
+        has_global_views = features.has(
+            "organizations:global-views", organization, actor=request.user
+        )
+        if not has_global_views and len(filter_params.get("project_id", [])) > 1:
+            raise ParseError(detail="You cannot view events from multiple projects.")
+
+        return filter_params
+
+    def get(self, request: Request, organization: Organization) -> Response:
+        if not features.has("organizations:session-replay", organization, actor=request.user):
+            return Response(status=404)
+        try:
+            filter_params = self.get_replay_filter_params(request, organization)
+        except NoProjects:
+            return Response({"data": []}, status=200)
+
+        result = ReplayValidator(data=request.GET)
+        if not result.is_valid():
+            raise ParseError(result.errors)
+
+        for key, value in result.validated_data.items():
+            if key not in filter_params:
+                filter_params[key] = value
```

Comment:
what can the user query for besides the time ranges and project? I guess can they filter this data by browser type, etc?

Reply:
Currently I've not added any filtering other than stats-period and project_id.  So it could be removed.

Reply:
I replaced ReplayValidator with a selector validator.  The supported fields are listed on the instance.